content/en/serverless/google_cloud_run/functions_gen1.md

```diff
@@ -0,0 +1,267 @@
+
+
+2. **Initialize the Datadog Java tracer and serverless compatibility layer**. Add `JAVA_TOOL_OPTIONS` to your runtime environment variable:
+   
+   [Auto instruments][1], the java tracer by setting the Runtime environment variable to wrap your Java cloud function with the Datadog Java tracer and the serverless compatibility layer.
+
+   | Name      | Value |
+   |-----------| ----- |
+   | `JAVA_TOOL_OPTIONS` | `-javaagent:dd-serverless-compat-java-agent.jar -javaagent:dd-java-agent.jar ` |
+
+3. (Optional) **Enable runtime metrics**. See [Java Runtime Metrics][2].
+
+4. (Optional) **Enable custom metrics**. See [Metric Submission: DogStatsD][3].
+
+[1]: /tracing/trace_collection/automatic_instrumentation/dd_libraries/java/?tab=wget
+[2]: /tracing/metrics/runtime_metrics/?tab=java
+[3]: /metrics/custom_metrics/dogstatsd_metrics_submission/?code-lang=java
+{{< /programming-lang >}}
+{{< /programming-lang-wrapper >}}
+
+5. **Deploy your function**. Utilize `gcloud` or Google Console to dpeloy your cloud function gen 1:
+
+   Follow this [Google Cloud Doc][10] to utilize `gcloud function deploy <FUNCTION_NAME> --no-gen2` to deploy a 1st Gen Cloud Run Function.
+
+   Utilize the `--source` flag to point to the directory of your function code with `dd-java-agent.jar` and `dd-serverless-compat-java-agent.jar` at the top level.
+
+   For more information read [this Google doc][11] for more flags for the gcloud command. 
+
+
+6. **Configure Datadog intake**. Add the following environment variables to your function's application settings:
+
+   | Name | Value |
+   | ---- | ----- |
+   | `DD_API_KEY` | Your [Datadog API key][1]. |
+   | `DD_SITE` | Your [Datadog site][2]. For example, {{< region-param key=dd_site code="true" >}}. |
+
+7. **Configure Unified Service Tagging**. You can collect metrics from your Cloud Run Function by installing the [Google Cloud integration][6]. To correlate these metrics with your traces, first set the `env`, `service`, and `version` tags on your resource in GC. Then, configure the following environment variables. You can add custom tags as `DD_TAGS`.
+
+   | Name | Value |
+   | ---- | ----- |
+   | `DD_ENV` | How you want to tag your env for [Unified Service Tagging][9]. For example, `prod`. |
+   | `DD_SERVICE` | How you want to tag your service for [Unified Service Tagging][9].  |
+   | `DD_VERSION` | How you want to tag your version for [Unified Service Tagging][9]. |
+   | `DD_TAGS` | Your comma-separated custom tags. For example, `key1:value1,key2:value2`.  |
+
+8. **Add Service Label**. Tag your GCP entity with the `service` label to correlate your traces with your service:
+
+   Add the same value from `DD_SERVICE` to a `service` label on your cloud function's inside the info panel of your function. More information on how to add labels can be found [here][12].
```

Comment:
```suggestion
   Add the same value from `DD_SERVICE` to a `service` label on your cloud function, inside the info panel of your function. For more information on how to add labels, see Google Cloud's [Configure labels for services][12] documentation.
```
I've been going through and replacing link text to make it less generic, which is really helpful for users who are using screen readers. If they're tabbing through links to find a specific one, descriptive link text is a huge help 🙂

Reply:
Of course!!